app/code/Magento/InventorySales/Model/ResourceModel/IsStockItemSalableCondition/BackordersCondition.php

```diff
@@ -35,11 +35,14 @@ public function __construct(StockConfigurationInterface $configuration)
     public function execute(Select $select): string
     {
         $globalBackorders = (int)$this->configuration->getBackorders();
+        $itemBackordersCondition = 'legacy_stock_item.backorders <> 0';
+        $useDefaultBackorders = 'legacy_stock_item.use_config_backorders';
+        $itemMinQty = 'legacy_stock_item.min_qty';
 
-        $condition = (1 === $globalBackorders)
-            ? 'legacy_stock_item.use_config_backorders = 1'
-            : 'legacy_stock_item.use_config_backorders = 0 AND legacy_stock_item.backorders = 1';
-        $condition .= ' AND (legacy_stock_item.min_qty >= 0 OR legacy_stock_item.qty > legacy_stock_item.min_qty)';
+        $condition = $globalBackorders === 0
+            ? "$useDefaultBackorders = 0 AND $itemBackordersCondition"
+            : "$useDefaultBackorders = 1 OR $itemBackordersCondition";
```

Comment:
How is additional `OR` clause used here? I believe the only problem was that `backorders` may be not only `0` and `1` but `2` as well, it seems to be covered by updating the `AND` clause above

Reply:
**OR** covers next case : 
* in M2 admin general settings _backorder_ is 1 or 2;
* on product _backorder_ is also 1 or 2 and 'Use default' for _backorder_ is unchecked;
* finally product won't present.